src/app/teams/teams-view.component.ts

```diff
@@ -92,8 +92,10 @@ export class TeamsViewComponent implements OnInit, AfterViewChecked, OnDestroy {
   ngAfterViewChecked() {
     const activeTab: MatTab = this.getActiveTab(this.initTab);
     if (activeTab && activeTab.position !== 0) {
-      setTimeout(() => this.tabSelectedIndex = this.tabSelectedIndex + activeTab.position, 0);
-      this.initTab = '';
+      setTimeout(() => {
+        this.tabSelectedIndex = this.tabSelectedIndex + activeTab.position;
+        this.initTab = activeTab.position === 0 ? '' : this.initTab;
```

Comment:
Hi, Paul, the `activeTab.position` will not never be 0 here since we have if condition above

Reply:
`tabSelectedIndex` modifies the `activeTab.position`.  For whatever reason the first time modifying this the `activeTab` is not selected, so adding that condition means it changes the `tabSelectedIndex` on the next change cycle also.

Reply:
Got it, thanks, so it will check whether the tab successfully changed to the desire tab. What will happen if not successfully changed, will it trigger some functions? Why do you make `initTab` to be `''` instead of `undefined`?

Reply:
`ngAfterViewChecked` runs every change cycle.  The `setTimeout` is a way of making the function within wait until the cycle completes, then changing `tabSelectedIndex` will start another.  Once the tab has successfully been selected, we change it to `''` so it only selects the tab when the component loads.  We run `setStatus` on initialization and after any changes the user makes to the team.  The condition on line 180 checks for `initTab === undefined` so it only gets set to a different value when the component is initializing.